packages/kolme/src/core/types/error.rs

```diff
@@ -19,6 +25,227 @@ pub enum KolmeError {
+    },
+
+    #[error("Executed block height mismatch: expected {expected}, got {actual}")]
+    ExecutedHeightMismatch {
+        expected: BlockHeight,
+        actual: BlockHeight,
+    },
+
+    #[error("Submitter error: {0}")]
+    Submitter(#[from] SubmitterError),
+
+    #[error("Import/export error: {0}")]
+    ImportExport(#[from] KolmeImportExportError),
+
+    #[error("Types error: {0}")]
+    TypesError(#[from] KolmeTypesError),
+
+    #[error("Core error: {0}")]
+    CoreError(#[from] KolmeCoreError),
+
+    #[error("Listener error: {0}")]
+    ListenerError(#[from] KolmeListenerError),
+
+    #[error("Execution error: {0}")]
+    ExecuteError(#[from] KolmeExecuteError),
+
+    #[error("Execution error: {0}")]
+    Execution(#[from] KolmeExecutionError),
+
+    #[error("Store error: {0}")]
+    StoreError(#[from] KolmeStoreError),
+
+    #[error("Failed to serialize Solana payload to Borsh: {details}")]
+    SolanaPayloadSerializationError { details: String },
+
+    #[error("Failed to build Solana initialization transaction: {details}")]
+    SolanaInitTxBuildFailed { details: String },
+
+    #[error("Solana submitter failed to execute signed transaction: {details}")]
+    SolanaSignedTxExecutionFailed { details: String },
+
+    #[error("Failed to create Solana pubsub client: {0}")]
+    SolanaPubsubError(String),
+
+    #[error("Bridge program {program} hasn't been initialized yet")]
+    UninitializedSolanaBridge { program: String },
+
+    #[error("Error deserializing Solana bridge state: {details}")]
+    InvalidSolanaBridgeState { details: String },
+
+    #[error("Error deserializing Solana bridge message from logs: {details}")]
+    InvalidSolanaBridgeLogMessage { details: String },
+
     #[error("{0}")]
     Other(String),
 }
+
+macro_rules! impl_from_to_other {
+    ($($source:ty => $msg:expr),+ $(,)?) => {
+        $(
+            impl From<$source> for KolmeError {
+                fn from(e: $source) -> Self {
+                    KolmeError::Other(format!($msg, e))
+                }
+            }
+        )+
+    };
+}
+
+impl_from_to_other! {
+    AddressError => "Address error: {}",
+    WalletError => "Wallet error: {}",
+    ParsePubkeyError => "Parse pubkey error: {}",
+    CoreStateError => "CoreState error: {}",
+    AssetError => "Asset error: {}",
+    AccountsError => "Accounts error: {}",
+    ValidatorSetError => "Validator set error: {}",
+    PublicKeyError => "Public key error: {}",
+    MerkleSerialError => "Merkle serialization error: {}",
+    TryFromIntError => "TryFromInt error: {}",
+    serde_json::Error => "JSON error: {}",
+    RecvError => "Recv error: {}",
+    cosmos::Error => "Cosmos error: {}",
+    base64::DecodeError => "Base64 decode error: {}",
+    solana_client::client_error::ClientError => "Solana client error: {}",
+    tokio_tungstenite::tungstenite::Error => "WebSocket error: {}",
+    std::io::Error => "IO error: {}",
+}
+
+impl From<anyhow::Error> for KolmeError {
```

Comment:
Can `impl From<anyhow::Error>` be removed? Ideally, any uses of anyhow in downstream errors should be replaced with concrete errors.